### PR TITLE
refactoring

### PR DESCRIPTION
**Pull request by MaksOnofriychuk**

### CODE REVIEW

File: src/components/Table/Columns.tsx [hunk 1]
```diff
@@ -10,46 +10,59 @@ import {
 } from "../../types/table";
 import "./columns.scss";
 
-export const Cell: FC<CellProps> = ({ id, value, testArr }) => {
-  const { addCountCell, hoverCell } = useActions();
+export const Cell: FC<CellProps> = ({ cell }) => {
+  const { addCountCell, setHoverCell } = useActions();
   const [isHovering, setIsHovering] = React.useState<boolean>(false);
+  const { hoverACells } = useSelectorHook((state) => state.table);
```
Comment (apiloian) [inline src/components/Table/Columns.tsx:42]: name

File: src/components/Table/Columns.tsx [hunk 2]
```diff
@@ -10,46 +10,59 @@ import {
...
     }
   }, [isHovering]);
 
-  const k = testArr.some((elem) => elem === value);
+  const isHover = hoverACells.some((elem) => elem.id === cell.id);
 
   return (
     <td
-      className={k ? "ok" : ""}
+      className={isHover ? "ok" : ""}
```
Comment (apiloian) [inline src/components/Table/Columns.tsx]: class name maybe better name selected or hovered

File: src/components/Table/Columns.tsx [hunk 3]
```diff
@@ -10,46 +10,59 @@ import {
...
+
+  const handleMouseOver = () => {
+    setIsHoverSum(true);
+  };
+
+  const handleMouseOut = () => {
+    setIsHoverSum(false);
+  };
+
+  console.log(isHoverSum);
```
Comment (apiloian) [inline src/components/Table/Columns.tsx]: any console.log

File: src/store/reducers/tableReducer.ts [hunk 4]
```diff
@@ -84,12 +94,39 @@ export default function tableReducer(
...
-    case TableActionsTypes.HOVERING_CELL:
-      console.log(action.payload);
+    case TableActionsTypes.SET_HOVERING_CELL:
+      const rowsValue: MatrixCell[][] = action.payload.value
+        ? state.matrix.map((row: MatrixCells) => row.value)
+        : [];
+      const cells: MatrixCell[] = rowsValue.flat(1);
+
+      const diferenceValue: MatrixCell[] = cells.map((cell: MatrixCell) => {
+        const aDiference = Math.abs(cell.value - action.payload.value);
```
Comment (apiloian) [inline src/store/reducers/tableReducer.ts]: aDiference not valid name

File: src/store/reducers/tableReducer.ts [hunk 5]
```diff
@@ -84,13 +95,50 @@ export default function tableReducer(
         matrix: newMatrix,
       };
 
-    case TableActionsTypes.HOVERING_CELL:
-      console.log(action.payload);
+    case TableActionsTypes.SET_HOVERING_CELL:
+      const rowsValue: MatrixCell[][] = action.payload.value
```
Comment (apiloian) [inline src/store/reducers/tableReducer.ts:99]: [][] => []

File: src/types/table.ts [hunk 6]
```diff
@@ -17,21 +17,25 @@ export type FormState = {
 export interface MatrixState {
   rows: number[];
   columns: number[];
-  formData: string | object;
-  matrix: MatrixCells[];
-  cellHover: number;
+  formData: string | FormState;
+  matrix: any[];
```
Comment (apiloian) [inline src/types/table.ts:21]: why any ?

**Review by arturiop — APPROVED** [review]

**Pull request merged**